### PR TITLE
Fix not awaiting the AudioSystem.arrayBufferDigest promise

### LINKED ISSUE #523: jpod101 invalid audio added to cards
- Add `{audio}` marker to field values
- Scan 思いやられる
- Add expression card
- Review/preview card
- Hear this: https://assets.languagepod101.com/dictionary/japanese/audiomp3.php?kanji=%E6%80%9D%E3%81%84%E3%82%84%E3%82%89%E3%82%8C%E3%82%8B&kana=%E3%81%8A%E3%82%82%E3%81%84%E3%82%84%E3%82%89%E3%82%8C%E3%82%8B

I didn't check if this was caused by audio/anki refactoring in these PRs, but they could have caused this:
https://github.com/FooSoft/yomichan/pull/399
https://github.com/FooSoft/yomichan/pull/401
https://github.com/FooSoft/yomichan/pull/427
https://github.com/FooSoft/yomichan/pull/442
https://github.com/FooSoft/yomichan/pull/477
https://github.com/FooSoft/yomichan/pull/496

### PR DESCRIPTION
**Pull request by toasted-nutbread**
Fixes #523

### CODE REVIEW

**Comment by siikamiika** [comment]
Uh oh, I think the empty `[sound:]` that's added now breaks Anki somehow (related: https://github.com/FooSoft/yomichan/pull/477#discussion_r417634675). Anki started playing everything I had there when I added a card for 思いやられる on this branch.

File: ext/mixed/js/audio-system.js [hunk 1]
```diff
@@ -195,8 +195,8 @@ class AudioSystem {
         );
     }
 
-    _isAudioBinaryValid(arrayBuffer) {
-        const digest = AudioSystem.arrayBufferDigest(arrayBuffer);
+    async _isAudioBinaryValid(arrayBuffer) {
+        const digest = await AudioSystem.arrayBufferDigest(arrayBuffer);
         switch (digest) {
```
Comment (siikamiika) [inline ext/mixed/js/audio-system.js:200]: Wish JS had types :sob:

**Comment by toasted-nutbread** [comment]
> Anki started playing everything I had there

What do you mean by this?

**Comment by siikamiika** [comment]
> > Anki started playing everything I had there
> 
> What do you mean by this?

I mean that it played all my audio files sequentially. Try adding some other cards with audio and open one with `[sound:]` and it should happen.

**Comment by siikamiika** [comment]
Were you able to reproduce the bug in your environment? I can report it to Anki if I find a proper channel for it. They seem to have disabled GitHub issues. Maybe they expect people to open a PR instead of reporting bugs.

**Comment by toasted-nutbread** [comment]
> They seem to have disabled GitHub issues. 

I noticed that too as I was looking at their source in the past.

I wasn't able to reproduce on Windows, though I didn't try extensively.
```
Version 2.1.22 (0ecc189a)
Python 3.8.0 Qt 5.14.1 PyQt 5.14.1
```

**Comment by siikamiika** [comment]
Ok. I'll switch to my Win10 VM and see if I can reproduce it using mpv.

**Comment by siikamiika** [comment]
I've created a PR to Anki fixing the issue: https://github.com/ankitects/anki/pull/612

**Pull request merged**